action.sh

```diff
@@ -15,11 +15,11 @@ fi
 # Fetch all K3s versions usable for the specified partial version.
 # Even if the version is specific and complete, assume it is possibly partial.
 url="${GITHUB_API_URL}/repos/${REPO}/releases?per_page=999"
-curl --silent --fail --location "${authz[@]}" "$url" | \
-  jq '.[] | select(.prerelease==false) | .tag_name' >/tmp/versions.txt
+releases=$(curl --silent --fail --location "${authz[@]-}" "$url")
+versions=$(jq <<< "$releases" '.[] | select(.prerelease==false) | .tag_name')
 
 echo "::group::All available K3s versions (unsorted)"
-cat /tmp/versions.txt
+echo "$versions"
```

Comment:
May I suggest `cat` here and below? `echo` squashes all versions into one line, while `cat` keeps them one per line.
```suggestion
cat <<< "$versions"
```

Reply:
> `echo` squashes all versions into one line, while `cat` keeps them one per line.

The variable/value contains newlines and double quotes preserves them.

```
● ./action.sh 
::group::All available K3s versions (unsorted)
"v1.20.5+k3s1"
"v1.19.9+k3s1"
"v1.18.17+k3s1"
"v1.20.4+k3s1"
…
```

e.g.

```
● v=$(cat /etc/os-release)

● echo $v
PRETTY_NAME="Debian GNU/Linux 10 (buster)" NAME="Debian GNU/Linux" VERSION_ID="10" VERSION="10 (buster)" VERSION_CODENAME=buster ID=debian HOME_URL="https://www.debian.org/" SUPPORT_URL="https://www.debian.org/support" BUG_REPORT_URL="https://bugs.debian.org/"

● echo "$v"
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
```

Reply:
Oh! Interesting. Indeed, in Ubuntu images, it echoes them one per line. It is something different in MacOS's Bash then, which squashes them in `echo` into a single line.

Reply:
I get newlines on macOS, too. 🤔 I'm pretty sure any squashing comes from word splitting when not using quotes.

```
● /bin/bash --norc

The default interactive shell is now zsh.
To update your account to use zsh, please run `chsh -s /bin/zsh`.
For more details, please visit https://support.apple.com/kb/HT208050.

bash-3.2$ echo "$BASH_VERSION"
3.2.57(1)-release

bash-3.2$ v=$(head /etc/rpc)

bash-3.2$ echo $v
# # $FreeBSD$ # rpc 88/08/01 4.0 RPCSRC; from 1.12 99/07/25 SMI # rpcbind 100000 portmap sunrpc rpcbind rstatd 100001 rstat rstat_svc rup perfmeter rusersd 100002 rusers nfs 100003 nfsprog ypserv 100004 ypprog mountd 100005 mount showmount

bash-3.2$ echo "$v"
#
# $FreeBSD$
# rpc 88/08/01 4.0 RPCSRC; from 1.12   99/07/25 SMI
#
rpcbind         100000  portmap sunrpc rpcbind
rstatd          100001  rstat rstat_svc rup perfmeter
rusersd         100002  rusers
nfs             100003  nfsprog
ypserv          100004  ypprog
mountd          100005  mount showmount
```